swapper-org/NodeChain
# Mayor fixes and updates

# Description



- Fixes #47

`Bitcoin-core`, `Electrum` and `Electron-Cash` has been updated to the latest version.


- Fixes #50

This was solved by downgrading the version of Electrumx, removing its files and rebuilding the container.
In the same PR we've decided to replace Electrumx to Electrs.

- Fixes #13 
- [x] Dockerfile for testnet and mainnet
- [x] Dockerfile for development
- [x] Entrypoints
- [x] Change docker-compose
- [x] Fit new changes into the other containers

## Dependencies (if any)

`Bitcoin-core` -> 0.21.0
`Electron-Cash` -> 4.2.6


## Type of change




# How Has This Been Tested?





## Bitcoin-Core update
1. Download bitcoin-core code from the official website
![imagen](https://user-images.githubusercontent.com/75782378/148068084-93e60020-9793-4055-9a36-3cc67f1be9ea.png)

2. Check generate hash locally
```sh
~$ sha256sum bitcoin-22.0-x86_64-linux-gnu.tar.gz 

59ebd25dd82a51638b7a6bb914586201e67db67b919b2a1ff08925a7936d1b16  bitcoin-22.0-x86_64-linux-gnu.tar.gz
```

3. Check `SHA256SUMS` file from the official website and compare with the local Hash
```sh
~$ cat SHA256SUMS 
9547fa03574f8bde296f707c7d9f7d89827c75c5a28f84402578a4fa92a787ec  bitcoin-22.0-aarch64-linux-gnu-debug.tar.gz
ac718fed08570a81b3587587872ad85a25173afa5f9fbbd0c03ba4d1714cfa3e  bitcoin-22.0-aarch64-linux-gnu.tar.gz
80071e0ecd24edfec8a1972b495b9822c79a5d33c7123bff51688638aac97cab  bitcoin-22.0-arm-linux-gnueabihf-debug.tar.gz
b8713c6c5f03f5258b54e9f436e2ed6d85449aa24c2c9972f91963d413e86311  bitcoin-22.0-arm-linux-gnueabihf.tar.gz
....

--> 59ebd25dd82a51638b7a6bb914586201e67db67b919b2a1ff08925a7936d1b16  bitcoin-22.0-x86_64-linux-gnu.tar.gz
....
ecc579d006230d6ffc5a5b7b53ce8c76477d37c1c7bad69694e9c2d69f00331d  bitcoin-22.0-win64-setup-unsigned.exe
9485e4b52ed6cebfe474ab4d7d0c1be6d0bb879ba7246a8239326b2230a77eb1  bitcoin-22.0-win64.zip
```

4. Check if Docker build the image:
```sh
scripts ~$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED              STATUS              PORTS                                                                            NAMES
925efe019b43   electrum               "/usr/src/app/entryp…"   About a minute ago   Up About a minute   30000/tcp                                                                        btc_testnet_api_electrum_1
c7ce676ca36e   lukechilds/electrumx   "init"                   About a minute ago   Up About a minute   8000/tcp, 50001-50002/tcp, 50004/tcp                                             btc_testnet_api_electrumx_1
0c43e0b6b5ac   bitcoincore            "/entrypoint.sh -rpc…"   About a minute ago   Up About a minute   8332-8333/tcp, 18332-18333/tcp, 18443-18444/tcp, 28332/tcp, 38332-38333/tcp      btc_testnet_api_bitcoincore_1
67edb21bcd15   connector              "/usr/src/app/Connec…"   About a minute ago   Up About a minute   80/tcp, 28332/tcp                                                                btc_testnet_api_connector_1
3f742a1b788d   nginx                  "/docker-entrypoint.…"   About a minute ago   Up About a minute   0.0.0.0:443->443/tcp, :::443->443/tcp, 0.0.0.0:60001->80/tcp, :::60001->80/tcp   btc_testnet_api_nginx_1
```

## Test Electrum update
1. Download Electrum code from the official website
![imagen](https://user-images.githubusercontent.com/75782378/148069515-0f431d43-f464-447e-add8-e4ec0a3d651a.png)

2. Check generate hash locally
```sh
~$ sha256sum Electrum-4.1.5.tar.gz 

91472e281f604a7a687faa4b339bf5e484bd2bec82a495ea45bfa014e52719a1  Electrum-4.1.5.tar.gz
```

3. Check if Docker build the image:
```sh
scripts ~$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED              STATUS              PORTS                                                                            NAMES
925efe019b43   electrum               "/usr/src/app/entryp…"   About a minute ago   Up About a minute   30000/tcp                                                                        btc_testnet_api_electrum_1
c7ce676ca36e   lukechilds/electrumx   "init"                   About a minute ago   Up About a minute   8000/tcp, 50001-50002/tcp, 50004/tcp                                             btc_testnet_api_electrumx_1
0c43e0b6b5ac   bitcoincore            "/entrypoint.sh -rpc…"   About a minute ago   Up About a minute   8332-8333/tcp, 18332-18333/tcp, 18443-18444/tcp, 28332/tcp, 38332-38333/tcp      btc_testnet_api_bitcoincore_1
67edb21bcd15   connector              "/usr/src/app/Connec…"   About a minute ago   Up About a minute   80/tcp, 28332/tcp                                                                btc_testnet_api_connector_1
3f742a1b788d   nginx                  "/docker-entrypoint.…"   About a minute ago   Up About a minute   0.0.0.0:443->443/tcp, :::443->443/tcp, 0.0.0.0:60001->80/tcp, :::60001->80/tcp   btc_testnet_api_nginx_1
```


## Test Electrum-Cash update
1. Download Electrum-Cash code from the official website
![imagen](https://user-images.githubusercontent.com/75782378/148071111-f2779116-4136-4fd4-9a30-310c29381584.png)

2. Check generate hash locally
```sh
sha256sum Electron-Cash-4.2.6.tar.gz 

3fbcb81c8efc005450595cbff975e44ebb77473e2b14ca27f6863efbf96b2f91  Electron-Cash-4.2.6.tar.gz
```



**Test Configuration**:
* Operating system (output of `cat /etc/os-release`):
```sh
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
* Kernel version (output of `uname -sr`): Linux 4.15.0-143-generic
* Architecture (output of `uname -m`): x86_64

# Related PR or Docs PR


Docs PR related # 
Other PR related # 

# Good practices to consider
